### PR TITLE
Preventing folder to be created before all prompts are filled

### PR DESCRIPTION
**Pull request by diego-antonelli**
Removing ESLint exceptions from preview file

### CODE REVIEW

**Review by Andries-Smit — APPROVED** [review]

File: generators/app/index.js [hunk 1]
```diff
@@ -32,13 +32,11 @@ class MxGenerator extends Generator {
          * Test if a widget folder was passed by argument and try to create the new folder
          */
         if(args.length > 0){
-            const dir = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toUpperCase())).join("");
+            const dir = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toLowerCase())).join("-");
+            const name = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toUpperCase())).join("");
```
Comment (Andries-Smit) [inline generators/app/index.js:36]: IMO: the name in the package.json name should be qual to the dir name.

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: generators/app/index.js (hunk 1 above)
Comment (diego-antonelli) [reply generators/app/index.js:36]: Fixed

**Review by mmehmetAliIzci — DISMISSED** [review]

File: generators/app/index.js [hunk 2]
```diff
@@ -48,12 +46,17 @@ class MxGenerator extends Generator {
...
             this.FINISHED = true;
             done();
-        } else {
+        }else if(this.dir && fsExtra.existsSync(this.dir) && !extfs.isEmptySync(this.dir)){
+            this.log(banner);
+            this.log(text.DIR_NOT_EMPTY_ERROR);
+            this.FINISHED = true;
+            done();
+        }else{
             done();
```
Comment (wegiswes) [inline generators/app/index.js]: Remove line 53, 58, 59, 61 and the code will be the same

File: generators/app/index.js [hunk 3]
```diff
@@ -383,6 +386,12 @@ class MxGenerator extends Generator {
 
     writing() {
         if(this.props) {
+            if(this.dir) {
```
Comment (wesleysieses) [inline generators/app/index.js]:
```suggestion
            if(this.dir && !fsExtra.existsSync(this.dir)) fsExtra.mkdirSync(this.dir);
            else if (this.dir) this.destinationRoot(this.dir);
```

File: generators/app/index.js [hunk 4]
```diff
@@ -383,6 +386,12 @@ class MxGenerator extends Generator {
...
+                if (!fsExtra.existsSync(this.dir)) {
+                    fsExtra.mkdirSync(this.dir);
+                }
+                this.destinationRoot(this.dir);
+            }
```
Comment (wesleysieses) [inline generators/app/index.js]:
```suggestion-5
if(this.dir && !fsExtra.existsSync(this.dir)) fsExtra.mkdirSync(this.dir);
else if (this.dir) this.destinationRoot(this.dir);
```

**Review by Andries-Smit — APPROVED** [review]

**Pull request merged**